test/parallel/test-child-process-execfile.js

```diff
@@ -99,3 +99,20 @@ const execOpts = { encoding: 'utf8', shell: true };
   });
   execFile(process.execPath, [fixture, 0], { signal }, callback);
 }
+
+// Verify the execFile() stdout is the same as execFileSync().
+{
+  const file = 'echo';
+  const args = ['foo', 'bar'];
+
+  // Test with and without `{ shell: true }`
+  [null, { shell: true }].forEach((options) => {
+    const execFileSyncResult = execFileSync(file, args, options);
+    assert.deepStrictEqual(execFileSyncResult, Buffer.from('foo bar\n'));
+
+    const check = common.mustCall((_, stdout) => {
+      assert.deepStrictEqual(Buffer.from(stdout), execFileSyncResult);
+    });
+    execFile(file, args, options, check);
```

Comment:
Nit: I think it would be more readable if we inline `check` function, `check` doesn't convey much readable information.
```suggestion
    execFile(file, args, options, common.mustCall((_, stdout) => {
      assert.deepStrictEqual(Buffer.from(stdout), execFileSyncResult);
    }));
```